app/scripts/views/behaviors/settings.js

```diff
@@ -34,9 +34,18 @@ define(function (require, exports, module) {
             if (options.success) {
               success = options.success;
             }
-            if (options.endpoint) {
-              endpoint = options.endpoint;
+
+            // Check the `redirectTo` param sent from server, if it matches
+            // a known path redirect there, else just go to settings.
+            const redirectTo = view.getSearchParam('redirectTo');
+            if (redirectTo) {
+              if (redirectTo.indexOf('/settings/emails') > 0) {
+                endpoint = '/settings/emails';
+              } else if (redirectTo.indexOf('/settings/two_step_authentication') > 0) {
+                endpoint = '/settings/two_step_authentication?showTwoStepAuthentication';
```

Comment:
Yea this definitely a hack. I added it since, the only way to get to the link is to have originally viewed the panel with the feature flag. 

I can try to move this logic into the auth broker.